src/sink/prometheus.rs

```diff
@@ -572,6 +572,14 @@ impl Sink for Prometheus {
         // nothing, intentionally
     }
 
+    fn shutdown(&mut self) -> () {
+        self.flush();
+        // This won't work:
+        // https://docs.rs/hyper/0.10.13/hyper/server/struct.Listening.html#method.close
```

Comment:
@pulltab here's the problem.

Reply:
Does the following line you introduced resolve the issue?

Reply:
No. :(